accounts-db/src/tiered_storage/hot.rs

```diff
@@ -430,6 +432,51 @@ impl HotStorageReader {
     }
 }
 
+/// The writer that creates a hot accounts file.
+#[derive(Debug)]
+pub struct HotStorageWriter {
+    storage: TieredStorageFile,
+}
+
+impl<'format> HotStorageWriter {
+    pub fn new(file_path: impl AsRef<Path>) -> TieredStorageResult<Self> {
+        Ok(Self {
+            storage: TieredStorageFile::new_writable(file_path)?,
+        })
+    }
+
+    /// An internal helper function that creates a new default footer
+    /// for hot accounts storage.
+    fn new_footer() -> TieredStorageFooter {
+        TieredStorageFooter {
+            account_meta_format: HOT_FORMAT.account_meta_format,
+            account_meta_entry_size: HOT_FORMAT.meta_entry_size as u32,
+            account_block_format: HOT_FORMAT.account_block_format,
+            index_block_format: HOT_FORMAT.index_block_format,
+            owners_block_format: HOT_FORMAT.owners_block_format,
+            ..TieredStorageFooter::default()
+        }
+    }
+
+    pub fn write_accounts<
+        'a,
+        'b,
+        T: ReadableAccount + Sync,
+        U: StorableAccounts<'a, T>,
+        V: Borrow<AccountHash>,
+    >(
+        &self,
```

Comment:
> An interesting API question, what happens after calling write_accounts()? Right now, it could be called twice. Should that be allowed? Will it break everything? Will the data be overwritten? Will there be two footers? What if the accounts written are different sizes the second time?

Good question.

The HotStorageWriter::new() is implemented using `TieredStorageFile::new_writable(file_path)`, which will fail to open an existing file as it opens the file with `create_new(true)`.  So one path can only be created once.

When it is used to implement `TieredStorage::append_accounts()`, this HotStorageWriter will only be created as a local variable inside `append_accounts()` temporarily.  So we won't be able to call write_accounts() twice as calling append_accounts() the second time will create a different HotStorageWriter (and using the same path will cause an Error on creation.)

Reply:
Added a test to verify creating a HotStorageWriter on an existing file will cause an error.